coralproject/talk
# [CORL-2339]: Add configuration for new commenter moderation across specific sites

## What does this PR do?

These changes add the ability to configure new commenter moderation for specific sites. These changes are for https://vmproduct.atlassian.net/browse/CORL-2339. For multi-site tenants, in the admin, under Moderation --> New commenter approval, there will now be `All sites`, `Specific sites`, and `Off` options, and the `Specific sites` option will allow specific sites to be searched for and selected.

## What changes to the GraphQL/Database Schema does this PR introduce?

In the schema, these changes:
- deprecate `premodEnabled` on settings --> newCommenters
- replace `premodEnabled` with `moderation` configuration on settings --> newCommenters. `moderation` includes `mode` for which moderation mode (`PRE`, `POST`, or `SPECIFIC_SITES_PRE`) should be applied to new commenters' comments, and it also includes `premodSites` which, for the `SPECIFIC_SITES_PRE` mode, includes any specific sites that should be pre-moderated.

- In models, these changes add `moderation` to `NewCommentersConfiguration` as well.

## Does this PR introduce any new environment variables or feature flags? 

no

## How do I test this PR?

You can test:
- changing settings in Configure in general, to ensure that this still works as expected
- changing `New commenter approval` settings under Configure --> Moderation. Testing this in both single-site and multi-site, since single site has only `On`/`Off` options.
- changing `Pre-moderate all comments` options under Configure --> Moderation. There was some refactoring related to this area since it's very similar to the updated `New commenter approval` settings.
- With different settings for new commenter approval (`All sites` and `Specific sites` with some sites selected and `Off`), create new commenters and attempt to add comments. See that comments are correctly either sent to pre-moderation until the number of comments that must be approved is reached, or they are sent through when moderation mode is set to `Off`/`POST`.
 
## How do we deploy this PR?
